src/math/sqrt_int16.c

```diff
@@ -0,0 +1,144 @@
+	int32_t xfi_tmp;
+	int32_t y;
+	uint16_t v;
+	int num_left_shifts;		//int8_t
+	int shift_factor;		//int8_t
+	int xfi;			//int8_t
+	unsigned int slice_temp;	//uint8_t
+	int sign = 1;
+	/*  Number of bits in a byte */
+	if ((int32_t)u == 0) {
+		y = 0;
+	} else {
+		/*  Normalize the input such that u = x * 2^n and 0.5 <= x < 2 */
+		/*  Normalize to the range [1, 2) */
+		/*  normalizes the input U */
+		/*  such that the output X is */
+		/*	  U = X*2^N */
+		/*	  1 <= X < 2 */
+		/*    The output X is unsigned with one integer bit. */
+		/*    The input U must be scalar and positive. */
+		/*    The number of bits in a byte is assumed to be B=8. */
+		/*  Reinterpret the input as an unsigned integer. */
+		/*  Unroll the loop in generated code so there will be no branching. */
+		/*  For each iteration, see how many leading zeros are in the high */
+		/*  byte of V, and shift them out to the left. Continue with the */
+		/*  shifted V for as many bytes as it has. */
+		/*  The index is the high byte of the input plus 1 to make it a */
+		/*  one-based index. */
+		/*  Index into the number-of-leading-zeros lookup table.  This lookup */
+		/*  table takes in a byte and returns the number of leading zeros in the */
+		/*  binary representation. */
+		shift_factor = iv[u >> 8ULL];
+		/*  Left-shift out all the leading zeros in the high byte. */
+		v = (u << shift_factor);
+		/*  Update the total number of left-shifts */
+		num_left_shifts = shift_factor;
+		/*  For each iteration, see how many leading zeros are in the high */
+		/*  byte of V, and shift them out to the left. Continue with the */
+		/*  shifted V for as many bytes as it has. */
+		/*  The index is the high byte of the input plus 1 to make it a */
+		/*  one-based index. */
+		/*  Index into the number-of-leading-zeros lookup table.  This lookup */
+		/*  table takes in a byte and returns the number of leading zeros in the */
+		/*  binary representation. */
+		shift_factor = iv[v >> 8ULL];
+		/*  Left-shift out all the leading zeros in the high byte. */
+		/*  Update the total number of left-shifts */
+		num_left_shifts += shift_factor;
+		/*  The input has been left-shifted so the most-significant-bit is a 1. */
+		/*  Reinterpret the output as unsigned with one integer bit, so */
+		/*  that 1 <= x < 2. */
+		/*  Let Q = int(u).  Then u = Q*2^(-u_fraction_length),	 */
+		/*  and x = Q*2^num_left_shifts * 2^(1-word_length).  Therefore, */
+		/*  u = x*2^n, where n is defined as: */
+		xfi_tmp = (3 - num_left_shifts) & 1;
+		v = ((v << shift_factor) >>
+			       xfi_tmp);
+		/*  Extract the high byte of x */
+		/*  Convert the high byte into an index for SQRTLUT */
+		slice_temp = SQRT_WRAP_SCHAR_BITS & (v >> 8ULL);
+		/*  The upper byte was used for the index into SQRTLUT. */
+		/*  The remainder, r, interpreted as a fraction, is used to */
+		/*  linearly interpolate between points. */
+		y = (iv1[((v >> 8ULL) - 63U) - 1] << 8ULL) +
```

Comment:
Do 63U, 194U have purpose here? I don't think you need unsigned integer wrap/overflow feature here. The DSP arithmetic is by default signed and unsigned may need extra instructions, so I'd use unsigned only when some unsigned wrap tricks are needed for code to work.